language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -56,8 +56,11 @@ main :: IO ()
 main = do
     opts@Options{..} <- execParser optionsParserInfo
     dar <- B.readFile optInputDar
-    dalfs <- either fail pure $ DAR.readDalfs $ Zip.toArchive $ BSL.fromStrict dar
-    forM_ ((DAR.mainDalf dalfs, optMainPackageName) : map (, Nothing) (DAR.dalfs dalfs)) $ \(dalf, mbPkgName) -> do
+    let archive = Zip.toArchive $ BSL.fromStrict dar
+    dalfs <- either fail pure $ DAR.readDalfs archive
+    DAR.DalfManifest{packageName, ..} <- either fail pure $ DAR.readDalfManifest archive
+    packageName <- pure $ optMainPackageName <|> packageName
```

Comment:
Whenever I see this pattern I wish Haskell's `let`s wouldn't be recursive by default. So sad. 😿